File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    
    ans = []
    for i in range(min(n, m), 0, -1):
        if n % i == 0 and m %i == 0:
            ans.append(i)
            break
    
    for i in range(max(n, m), n*m+1):
        if i % n == 0 and i % m == 0:
            ans.append(i)
            break
    
    return ans

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int[] ans = new int[2];
        
        ans = minNum(n, m);
        
        return ans;
    }
    
    static int[] minNum(int n, int m) {
       
        // switch
        int tmp = 0;
        if(n > m) {
            tmp = n;
            n = m;
            m = tmp;
        }
        
        int ans = 0;
        for(int i = m; i > 0; i--) {
            if (n % i == 0 && m % i == 0) {
                ans = i;
                break;
            }
        }
        
        return new int[]{ans, ans * (n/ans) * (m/ans)};
        
    }
   
}